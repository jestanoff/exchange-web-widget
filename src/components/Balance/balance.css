.balance {
  color: #888888;
  font-size: 14px;
  padding-left: 2px;
}
