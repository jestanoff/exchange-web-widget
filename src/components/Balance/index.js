import React from 'react';
import { oneOf, string } from 'prop-types';
import getCurrencyCodes from '../../utils/getCurrencyCodes';
import * as currencies from '../../constants/currencies';
import styles from './balance.css';

const Balance = ({ amount, currency }) => (
  <span className={styles.balance}>Balance: {`${currencies[currency].symbol}${amount}`}</span>
);

Balance.propTypes = {
  amount: string,
  currency: oneOf(getCurrencyCodes(currencies)),
};

Balance.defaultProps = {
  amount: '0',
};

export default Balance;
