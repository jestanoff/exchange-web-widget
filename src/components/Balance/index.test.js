import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Balance from './';

describe('Balance component', () => {
  test('should render proper markup', () => {
    const { container } = render(<Balance amount="99.10" currency="EUR" />);

    expect(container).toMatchSnapshot();
  });

  test('should show default amount of 0 if no amount has been passed', () => {
    const { getByText } = render(<Balance currency="USD" />);

    expect(getByText('Balance: $0')).toBeInTheDocument();
  });
});
