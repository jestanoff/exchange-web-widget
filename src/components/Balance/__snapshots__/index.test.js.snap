// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance component should render proper markup 1`] = `
<div>
  <span
    class="balance"
  >
    Balance: 
    â‚¬99.10
  </span>
</div>
`;
